cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -416,27 +420,32 @@ private void doPerformAction(FullHttpRequest request, HttpResponder responder, S
     ProgramType programType = getProgramType(type);
     ProgramId program = applicationId.program(programType, programId);
     Map<String, String> args = decodeArguments(request);
-    String latestVersion = getLatestAppVersion(new NamespaceId(namespaceId), appId);
     // we have already validated that the action is valid
     switch (action.toLowerCase()) {
       case "start":
-        if (!appVersion.equals(latestVersion)) {
-          throw new BadRequestException("start action is only allowed on the latest program version");
+        if (ApplicationId.DEFAULT_VERSION.equals(appVersion)) {
+          lifecycleService.run(new NamespaceId(namespaceId), appId, programType, programId, args, false);
+        } else {
+          lifecycleService.run(program, args, false);
```

Comment:
let's annotate this method in the ProgramLifecycleService as `@deprecated`, we don't want any new code to call this.